RIOT-OS/RIOT
# cpu/esp: cleanup of ESP SDK log outputs

### Contribution description

The ESP SDK libraries generate a lot of log outputs at the default log level, which are usually not of interest. Especially, the WiFi SDK libraries that are used by `esp_now` and `esp_wifi` produce a lot of such output about driver internals, for example
```
add if0
add if1
bcn 100
main(): This is RIOT! (Version: 2020.01-devel-1500-g31330-cpu/esp ...)
RIOT network stack example application
All up, running the shell now
> scandone
state: 0 -> 2 (b0)
state: 2 -> 3 (0)
state: 3 -> 5 (10)
add 0
aid 15
cnt 

connected with SSID, channel 8
WiFi connected to ssid SSID, channel 8
```

This PR implements a set of library-specific `printf` functions that intercept the log output from the SDK libraries and redirect them to the RIOT log macros. This allows to control the log output of the ESP SDK libraries according to the defined log level.

The PR contains also some small cleanups of log outputs in initialization functions.

**Note:** This PR requires a modified SDK version for ESP8266 boards to use the changes. The modified SDK version is available at [GitHub](https://github.com/gschorcht/RIOT-Xtensa-ESP8266-RTOS-SDK.git).

### Testing procedure

Test any ESP32 board with:
```
USEMODULE='esp_wifi' CFLAGS='-DESP_WIFI_SSID=\"pass\" -DESP_WIFI_PASS=\"ssid\"' \
make BOARD=esp32-wroom-32 -C examples/gnrc_networking flash term
```
```
LOG_LEVEL=4 USEMODULE='esp_wifi' CFLAGS='-DESP_WIFI_SSID=\"pass\" -DESP_WIFI_PASS=\"ssid\"' \
make BOARD=esp32-wroom-32 -C examples/gnrc_networking flash term
```
With `LOG_LEVEL=4`, you should see the log output as before. At default log level you should only see
```
main(): This is RIOT! (Version: 2020.01-devel-1500-g31330-cpu/esp ...)
RIOT network stack example application
All up, running the shell now
> WiFi connected to ssid SSID, channel 8
```
The test of the PR with an ESP8266 board requires a modified SDK Version. Without the modified version of the modified SDK Version, ESP8266 should be compilable. The modified version of the SDK can be installed with:
```
cd /path/to
git clone https://github.com/gschorcht/RIOT-Xtensa-ESP8266-RTOS-SDK.git
git checkout release/v3.1-for-riot-os-v2
export ESP8266_RTOS_SDK_DIR=/path/to/RIOT-Xtensa-ESP8266-RTOS-SDK
```
Once the modified SDK version is installed, you can use:
```
USEMODULE='esp_wifi' CFLAGS='-DESP_WIFI_SSID=\"pass\" -DESP_WIFI_PASS=\"ssid\"' \
make BOARD=esp8266-esp-12x -C examples/gnrc_networking flash term
```
```
LOG_LEVEL=4 USEMODULE='esp_wifi' CFLAGS='-DESP_WIFI_SSID=\"pass\" -DESP_WIFI_PASS=\"ssid\"' \
make BOARD=esp8266-esp-12x -C examples/gnrc_networking flash term
```

### Issues/PRs references

This PR is prerequisite for PR https://github.com/RIOT-OS/riotdocker/pull/95
